VideoLocker/src/main/java/org/edx/mobile/model/api/CourseEntry.java

```diff
@@ -248,22 +252,80 @@ public boolean isGroupAvailable(SocialUtils.SocialType type) {
 
     }
 
-    public String getDescription(Context context){
+    public String getDescription(Context context, boolean withStartDate) {
         StringBuilder detailBuilder = new StringBuilder();
-        if ( getOrg() != null){
-            detailBuilder.append( getOrg());
+
+        if (!TextUtils.isEmpty(org)) {
+            detailBuilder.append(org);
         }
-        if ( getNumber() != null) {
-            if (detailBuilder.length() > 0){
+
+        if (!TextUtils.isEmpty(number)) {
+            if (detailBuilder.length() > 0) {
                 detailBuilder.append(" | ");
             }
-            detailBuilder.append( getNumber());
+            detailBuilder.append(number);
 
         }
-        //https://openedx.atlassian.net/browse/MA-858
-        //we remove the ending data for now
+
+        if (withStartDate) {
+            if (detailBuilder.length() > 0) {
+                detailBuilder.append(" | ");
+            }
+
+            detailBuilder.append(getFormattedStartDate(context));
+        }
+
+        if (detailBuilder.length() > 0) {
+            int secondlastCharIndex = detailBuilder.length() - 2;
+            if (detailBuilder.charAt(secondlastCharIndex) == '|') {
+                detailBuilder.delete(secondlastCharIndex, detailBuilder.length());
+            }
+        }
 
         return detailBuilder.toString();
     }
 
+    @SuppressLint("SimpleDateFormat")
+    public String getFormattedStartDate(Context context){
+        StringBuilder dateBuilder = new StringBuilder();
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("MMMM dd");
```

Comment:
Seems like if you pass a locale to it as I suggested, should be fine, but it doesn't matter to me as long as we change this to use a localized version instead of just suppressing the lint warning